samigo/samigo-qti/src/java/org/sakaiproject/tool/assessment/qti/helper/ExtractionHelper.java

```diff
@@ -3254,4 +3257,19 @@ private Set<ItemTextAttachmentIfc> makeEMIItemTextAttachmentSet(ItemText itemTex
 		}
 		return attachSet;
 	}
+
+    public boolean updateSectionBasedOnQuestionPool(SectionFacade section, Map sectionMap) {
```

Comment:
Add type for Map, then you can remove the String cast on the get calls

Reply:
I prefer to stick with a tested version, this method is called after some operations over a map without type definition, it's probably safe but safer stick to a tested version.